workshops/painting_app/painting_app.md

```diff
@@ -0,0 +1,323 @@
+
+So let's start by creating a `div` container with the class `main-container` inside of our `<body>` tag. In this container we'll put all of our html elements.
+```html
+<div class="main-container"></div>
+```
+
+Then, we are creating another `div` container, that will hold all the color button. Inside of this `div` container, we'll put 6 `button` elements, with the following values:
+```html
+<div class="main-container">
+	<div class="colors">
+		<button value="#FBFF00"></button>
+		<button value="#009fff"></button>
+		<button value="#000000"></button>
+		<button value="#FF0000"></button>
+		<button value="#3EFF00"></button>
+		<button value="#FFF"></button>
+	</div>
+</div>
+```
+Each button represent its respective HEX code.
+
+Finally, right after the `div` container with the class `colors`, we'll put 4 elements with the following values:
+```
+<div class="main-container">
+	<div class="colors">
+		<button value="#FBFF00"></button>
+		<button value="#009fff"></button>
+		<button value="#000000"></button>
+		<button value="#FF0000"></button>
+		<button value="#3EFF00"></button>
+		<button value="#FFF"></button>
+	</div>
+
+   <--! Here we put the 4 elements-->
+
+	<input id="brush" type="range" value="1" min="1" max="5" step="1"></input>
+	<button id="clear" type="button">Clear</button>
+	<button id="save" type="button">Save</button>
+	<br>
+	<canvas id="paint-canvas" width="640" height="400"></canvas>
+
+</div>
+```
+Let's break this down:
+- The `input` element is used to create interactive controls for web-based forms in order to accept data from the user, since the type of input we are using is `range`, it requires you to set the attributes `value`, `min`, `max` and `step` which basically set the initial value for the input, the max and min value for it, and the step from a value to other respectively
+- The two buttons here will be used to clear and save the canvas data.
+- The `canvas` element is used to let the user make drawings through it.
+
+*Note: All the attributes and values used for the seen elements are mandatory in order to the app to work*
+
+Here's the result of the code above
+![Result](https://cloud-lblb3umex.vercel.app/0image.png)
+
+## JS
+
+Now that we have our html document ready to go, we are going to start by giving all the elements its functionalities.
+
+So, the first thing we want to do, is to use the `onload` event, which runs a block of code when a given resource has loaded. In this case, we want our javascript file to run when the html document has finished to load.
+```js
+windows.onload(){
+
+}
+```
+*Note: from now, all the javascript code will go inside of the curly braces*
+
+The next thing we are going to do is to define 4 variables:
+```js
+	var canvas = document.getElementById("paint-canvas");
+	var context = canvas.getContext("2d");
+	var boundings = canvas.getBoundingClientRect();
+	var range = document.getElementById("brush").value;
```

Comment:
try to avoid using `var` as it is not a good practice. use `let` / `const` instead. (make sure you do this change everywhere)